src/controller/CHIPDeviceController_deprecated.cpp

```diff
@@ -166,12 +166,15 @@ bool ChipDeviceController::IsConnected() const
     return mState == kState_Initialized;
 }
 
-bool ChipDeviceController::GetIpAddress(Inet::IPAddress & addr) const
+bool ChipDeviceController::GetIpAddress(Inet::IPAddress & addr)
 {
-    if (IsConnected() && mDevice != nullptr)
-        return mDevice->GetIpAddress(addr);
+    if (!IsConnected())
+        return false;
 
-    return false;
+    if (mDevice == nullptr)
```

Comment:
This sounds like an unusual sideffect - get ip addres may do a non-const init. Do we really need it?

or should we ignore since _deprecated should go away anyway?

Reply:
I admit it's not the most elegant solution, but I thought it didn't have to be since _deprecated will go away anyway ;). However, until it happens we should have code that works at least. Currently, it's impossible to test that Rendezvous completed successfully. Should I rename `getIpAddress` to something not suggesting that it's a simple getter? `retrieveIpAddress` mabe?

Reply:
I was just wondering about losing const.
Would if work if we return false if mDevice is nullptr?

That way we can stay const.

Reply:
No, mDevice is only initialized when sending a message/command and `getIpAddress` was introduced to automatically obtain IP address of a connected device before sending it a message to avoid typing long IPv6 addresses manually. The feature will probably be replaced with service discovery, but for now it's quite helpful I think.